apps/freeablo/engine/input.cpp

```diff
@@ -73,51 +89,59 @@ namespace Engine
 
         FAGui::Console & console = FAGui::Console::getInstance(FARender::Renderer::get()->getRocketContext());
 
-        if(hotkey == toggleconsole_key)
+        if(hotkey == getHotkey(TOGGLE_CONSOLE))
         {
-            mToggleConsole = true;;
+            mToggleConsole = true;
         }
         else if(!console.isVisible())
         {
-            if (hotkey == quit_key)
+            if (hotkey == getHotkey(QUIT))
             {
-                mEngine.stop();
+                notifyKeyboardObservers(QUIT);
             }
-            else if (hotkey == noclip_key)
+            else if (hotkey == getHotkey(NOCLIP))
             {
-                mEngine.toggleNoclip();
+                notifyKeyboardObservers(NOCLIP);
             }
-            else if (hotkey == changelvlup_key || hotkey == changelvldwn_key)
+            else if (hotkey == getHotkey(CHANGE_LEVEL_UP))
             {
-                FAWorld::World* world = FAWorld::World::get();
-                size_t nextLevelIndex;
-                if(hotkey == changelvlup_key)
-                    nextLevelIndex = world->getCurrentLevel()->getPreviousLevel();
-                else
-                    nextLevelIndex = world->getCurrentLevel()->getNextLevel();
-
-                world->setLevel(nextLevelIndex);
+                notifyKeyboardObservers(CHANGE_LEVEL_UP);
+            }
+            else if (hotkey == getHotkey(CHANGE_LEVEL_DOWN))
+            {
+                notifyKeyboardObservers(CHANGE_LEVEL_DOWN);
+            }
```

Comment:
or better: just use set of enums and simple check set.find(hotkey) != set.end() { notify(..) }